### PR TITLE
Updating the file frontend-interview-preps/JavaScript/Programmes/Symbol.js

### LINKED ISSUE #13: Issue in the "frontend-interview-preps/JavaScript/Programmes/Symbol.js" file
In the main branch, there are a few typos in the file, and the privatization of _password does not work.

### PR DESCRIPTION
**Pull request by Augasty**
Updated the code in the aforementioned file. 

Link to the issue: [link](https://github.com/ishwarrimal/frontend-interview-preps/issues/13)

### CODE REVIEW

File: JavaScript/Programmes/Symbol.js [hunk 1]
```diff
@@ -1,16 +1,17 @@
-const _password = Symbol("password");
 class User {
-  constructor(unsername, password) {
-    this.username = this.username;
-    this[_password] = password;
-  }
+  constructor(username, password) {
+  this.username = username;
+  let _password = Symbol(password)
```
Comment (ishwarrimal) [inline JavaScript/Programmes/Symbol.js:4]: This is suppose to be `this[_password]` as _password is a Symbol and we want to use the symbol here.

File: JavaScript/Programmes/Symbol.js [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
+
 
-  getPassword() {
-    return this[_password];
+  this.getPassword = () => {
```
Comment (ishwarrimal) [inline JavaScript/Programmes/Symbol.js:7]: this reference is not required here.

**Comment by ishwarrimal** [comment]
The example is quite not clear. I wrote this long back when I was doing something, which I can not recall why.
Would really appreciate if you can come up with a better example for Signal.

File: JavaScript/Programmes/Symbol.js (hunk 1 above)
Comment (Augasty) [reply JavaScript/Programmes/Symbol.js:4]: As far as I am aware, if we use this[_password] or this._password, it becomes accessible even from the outside of the User object. 
As our goal is to make the password only accessible via the getter function, I used **let _password**.

File: JavaScript/Programmes/Symbol.js (hunk 2 above)
Comment (Augasty) [reply JavaScript/Programmes/Symbol.js:7]: We won't be able to access the _password from outside the constructor, that's why the function is given as this.getPassword.

**Comment by Augasty** [comment]
Um, I did some digging, and Symbol is not at all suited for hashing or encrypting passwords... as there are no ways to decrypt it. 
We can use other hashing and salting methods for that.

File: JavaScript/Programmes/Symbol.js (hunk 1 above)
Comment (ishwarrimal) [reply JavaScript/Programmes/Symbol.js:4]: Symbol is used to create a unique variable name i.e. key for our object...in your case, you've created a unique value to the key, which is not the intended usage.
And a key created using a symbol is not accessible using . or [] notation.

File: JavaScript/Programmes/Symbol.js (hunk 2 above)
Comment (ishwarrimal) [reply JavaScript/Programmes/Symbol.js:7]: That's a syntax error. this method is not inside a constructor (not a static method) and you can't create an instance method using this.methodName in a class.

**Comment by ishwarrimal** [comment]
Hey, the _password field wasn't intended to be hashed or something....was just a key to an object.
 Symbol is used to create a unique variable name.
I just wanted a general example of it's usage.

**Comment by Augasty** [comment]
Ohh, I misunderstood and thought that you wanted it to be a hashing function of some sort. My bad, I will fix it.

**Comment by ishwarrimal** [comment]
No issues. Request for re-approval once you fix it.

**Comment by ishwarrimal** [comment]
Hey, any plans of updating this?
You could just fix the first line (this.username = username) for now and maybe raise another PR when you come up with a better example.

**Pull request closed without merging**